packages/fxa-payments-server/src/routes/Checkout/index.test.tsx

```diff
@@ -0,0 +1,141 @@
+  expectNockScopesDone,
+  defaultAppContextValue,
+  MockApp,
+  setupMockConfig,
+  mockConfig,
+  mockServerUrl,
+  mockOptionsResponses,
+} from '../../lib/test-utils';
+
+jest.mock('../../lib/sentry');
+
+jest.mock('../../lib/flow-event');
+
+import { SignInLayout } from '../../components/AppLayout';
+import Checkout from './index';
+import { AppContextType } from '../../lib/AppContext';
+import { PLANS } from '../../lib/mock-data';
+
+describe('routes/Checkout', () => {
+  let authServer = '';
+  let profileServer = '';
+
+  beforeEach(() => {
+    setupMockConfig({
+      ...mockConfig,
+      productRedirectURLs: PRODUCT_REDIRECT_URLS,
+    });
+    authServer = mockServerUrl('auth');
+    mockOptionsResponses(authServer);
+    profileServer = mockServerUrl('profile');
+    mockOptionsResponses(profileServer);
+  });
+
+  afterEach(() => {
+    noc.cleanAll();
+    return cleanup();
+  });
+
+  const Subject = ({
+    productId = PRODUCT_ID,
+    planId,
+    accountActivated,
+    matchMedia = jest.fn(() => false),
+    navigateToUrl = jest.fn(),
+    appContext = defaultAppContextValue(),
+  }: {
+    productId?: string;
+    planId?: string;
+    matchMedia?: (query: string) => boolean;
+    navigateToUrl?: (url: string) => void;
+    accountActivated?: string;
+    appContext?: Partial<AppContextType>;
+  }) => {
+    const props = {
+      match: {
+        params: {
+          productId,
+        },
+      },
+      plans: PLANS,
+      plansByProductId: () => PLANS[0],
+      fetchCheckoutRouteResources: () => PLANS,
+    };
+    const appContextValue = {
+      ...defaultAppContextValue(),
+      ...appContext,
+      matchMedia,
+      navigateToUrl: navigateToUrl || jest.fn(),
+      queryParams: {
+        plan: planId,
+        activated: accountActivated,
+      },
+    };
+    return (
+      <MockApp {...{ appContextValue }}>
+        <SignInLayout>
+          <Checkout {...props} />
+        </SignInLayout>
+      </MockApp>
+    );
+  };
+
+  const initApiMocks = () => [
+    nock(authServer)
+      .get('/v1/oauth/subscriptions/plans')
+      .reply(200, MOCK_PLANS, { 'Access-Control-Allow-Origin': '*' }),
+  ];
+
+  it('renders as expected', async () => {
+    const apiMocks = initApiMocks();
+    const { findByTestId } = render(<Subject />);
+    if (window.onload) {
+      dispatchEvent(new Event('load'));
+    }
+
+    const checkoutEle = await findByTestId('checkout-route-container');
+    expect(checkoutEle).toBeInTheDocument();
+    expectNockScopesDone(apiMocks);
+  });
+
+  it('displays an error with invalid product ID', async () => {
+    const apiMocks = initApiMocks();
+    const { findByTestId, queryByTestId } = render(
+      <Subject productId="bad_product" />
+    );
+    await findByTestId('no-such-plan-error');
+    expect(queryByTestId('dialog-dismiss')).not.toBeInTheDocument();
+    expectNockScopesDone(apiMocks);
+  });
+
+  it('displays an error on failure to load plans', async () => {
```

Comment:
This test case  and one other is identical to that in the `Product` route. Is there a way to make this DRYer?